src/pilot/assign_role.py

```diff
@@ -959,6 +959,11 @@ def generate_osd_config(ip_mac_service_tag, drac_client):
 def get_drives(drac_client):
     spinners = []
     ssds = []
+    # Getting all NVMe drives
+    nvme_drives = [drives for drives in drac_client.list_physical_disks()
+            if drives.device_protocol is not None  and
+            drives.device_protocol.startswith("NVMe")]
```

Comment:
Hey Pranjali, we always try to limit the number of calls we make to the iDRAC as much as possible.  There is a call to list_physical_disks() on line 963 above and also one on line 976 below when I think we could retrieve the physical disks just one time.  I think I would move this code into the "for" loop below, just below this line "for pd_id in physical_disks:".
You could check the protocol like you do above and if the drive is an NVMe drive, add it to an "nvmes" list, just like we do with spinners and ssds.

Reply:
Sure Chris. It will be incorporated as recommended.